packages/playground/src/weblets/tf_nextcloud.vue

```diff
@@ -0,0 +1,252 @@
+          #="{ index, isRequired }"
+          :required="[0]"
+        >
+          <input-validator
+            :value="envs[index].key"
+            :rules="[
+              validators.required('Key name is required.'),
+              key => validators.isAlpha('Key must start with alphabet char.')(key[0]),
+              validators.pattern('Invalid key format.', { pattern: /^[^0-9_\s][a-zA-Z0-9_]+$/ }),
+              validators.maxLength('Key max length is 128 chars.', 128),
+            ]"
+            #="{ props }"
+          >
+            <input-tooltip tooltip="Environment key.">
+              <v-text-field label="Name" v-model="envs[index].key" :disabled="isRequired" v-bind="props" />
+            </input-tooltip>
+          </input-validator>
+
+          <input-validator
+            :value="envs[index].value"
+            :rules="[validators.required('Value is required.')]"
+            #="{ props }"
+          >
+            <input-tooltip tooltip="Environment Value.">
+              <v-textarea label="Value" v-model="envs[index].value" no-resize :spellcheck="false" />
+            </input-tooltip>
+          </input-validator>
+        </ExpandableLayout>
+      </template>
+    </d-tabs>
+
+    <template #footer-actions>
+      <v-btn color="primary" variant="tonal" :disabled="tabs?.invalid || network?.error" @click="deploy">Deploy</v-btn>
+    </template>
+  </weblet-layout>
+</template>
+
+<script lang="ts" setup>
+import { computed, type Ref, ref } from "vue";
+
+import Network from "../components/networks.vue";
+import SelectSolutionFlavor from "../components/select_solution_flavor.vue";
+import { useLayout } from "../components/weblet_layout.vue";
+import { useProfileManager } from "../stores";
+import { type Farm, type Flist, ProjectName } from "../types";
+import { deployVM, type Disk, type Env } from "../utils/deploy_vm";
+import { getGrid } from "../utils/grid";
+import { generateName } from "../utils/strings";
+
+const layout = useLayout();
+const tabs = ref();
+const profileManager = useProfileManager();
+
+const images = [
+  {
+    name: "Nextcloud All-in-One",
+    flist: "https://hub.grid.tf/idrnd.3bot/logismosis-nextcloud-aio2-latest.flist",
+    entryPoint: "/sbin/zinit init",
+  },
+];
+
+const name = ref(generateName(8, { prefix: "nc" }));
+const flist = ref<Flist>();
+const ipv4 = ref(true);
+const planetary = ref(true);
+const wireguard = ref(true);
+const farm = ref() as Ref<Farm>;
+const envs = ref<Env[]>([]);
+const disks = ref<Disk[]>([]);
+const network = ref();
+const dedicated = ref(false);
+const certified = ref(false);
+const selectedNode = ref() as Ref<INode>;
+const rootFilesystemSize = 8;
+function layoutMount() {
+  if (envs.value.length > 0) {
+    envs.value.splice(0, 1);
+  }
+
+  envs.value.unshift({
+    key: "SSH_KEY",
+    value: profileManager.profile!.ssh,
+  });
+
+  envs.value.unshift({
+    key: "NEXTCLOUD_DOMAIN",
+    value: ipv4.value,
+  });
+}
```

Comment:
```suggestion
const name = ref(generateName(8, { prefix: "nc" }));
const flist = ref<Flist>();
const ipv4 = ref(true);
const planetary = ref(true);
const wireguard = ref(true);
const farm = ref() as Ref<Farm>;
const network = ref();
const dedicated = ref(false);
const certified = ref(false);
const selectedNode = ref() as Ref<INode>;
const rootFilesystemSize = 8;
```

We don't need to manage envs or disks